src/Symfony/Component/VarDumper/Cloner/AbstractCloner.php

```diff
@@ -69,6 +69,10 @@
         'Symfony\Component\DependencyInjection\ContainerInterface' => 'Symfony\Component\VarDumper\Caster\StubCaster::cutInternals',
         'Symfony\Component\VarDumper\Exception\ThrowingCasterException' => 'Symfony\Component\VarDumper\Caster\ExceptionCaster::castThrowingCasterException',
 
+        'PHPUnit_Framework_MockObject_MockObject' => 'Symfony\Component\VarDumper\Caster\StubCaster::cutInternals',
+        'PHPUnit_Framework_MockObject_Stub' => 'Symfony\Component\VarDumper\Caster\StubCaster::cutInternals',
+        'Prophecy\Prophecy\ProphecyInterface' => 'Symfony\Component\VarDumper\Caster\StubCaster::cutInternals',
```

Comment:
well, it depends where you are trying to dump things actually. Can you show a snippet of your code to see where you used dumping together with Prophecy ?
but in any case, ``Prophecy\Prophecy\ProphecyInterface`` is only implemented on pure Prophecy objects (i.e. not extending your code at all). I'm not sure it makes sense to cut all internals there.
Maybe we would only need to cut only the circular reference between the ObjectProphecy (implementing this interface) and the MethodProphecy when dumping.